commons/network/src/main/java/in/projecteka/jataayu/network/utils/RetrofitCallback.kt

```diff
@@ -0,0 +1,53 @@
+package `in`.projecteka.jataayu.network.utils
+
+import `in`.projecteka.jataayu.network.model.Error
+import `in`.projecteka.jataayu.network.model.ErrorResponse
+import androidx.lifecycle.MutableLiveData
+import com.google.gson.Gson
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.net.HttpURLConnection.HTTP_FORBIDDEN
+import java.net.HttpURLConnection.HTTP_UNAUTHORIZED
+
+private const val DEFAULT_ERROR_CODE = -1
+private const val ERROR_CODE_UNAUTHORIZED = 1017
+
+abstract class RetrofitCallback<T> : Callback<T?> {
```

Comment:
not required to mark generic Optional. While casting, it will anyway become an optional.

Reply:
Generics enforced me to make it nullable. But I didn't put much effort to remove that nullable part. Pls remove it if not required.